src/components/CountdownComponent/CountdownComponent.tsx

```diff
@@ -1,5 +1,50 @@
-function CountdownComponent() {
-  return <h1>Countdown component</h1>;
-}
+import React, { useState, useEffect } from 'react';
+
+// a functional component
+const CountdownComponent: React.FC = () => {
+  const targetDate = '2024-12-24T23:59:59'; // for now let's say 'time until christmas'
```

Comment:
Galvoju most likely geriau butu sita iskelti i env faila, kuriama kintamasis butu pavadinimu VITE_MIDI_DATE
https://vite.dev/guide/env-and-mode